rep/dao/RepresentativeInfoDao.py

```diff
@@ -0,0 +1,115 @@
+from rep.dao.BaseDao import BaseDao
+from rep.dataclasses.RepresentativeInfo import RepresentativeInfo
+
+class RepresentativeInfoDao(BaseDao):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.TABLE = "representative_info"
+        self.COLUMNS = ["id", "dist", "officeCode", "districtNumber", "number", 
+            "designatorCode", "firstName", "middleInitial", "lastName", "suffix", 
+            "commonlyUsedName", "homeStreetAddress", "homeCity", "homeState", 
+            "homeZipCode", "homePhone", "capitolStreetAddress", "capitolCity", 
+            "capitolPhone", "room", "roomNumber", "committeesChaired", 
+            "committeesViceChaired", "rankingMember", "committeeMember1", 
+            "senatorRepresentative", "party", "title", "gender", "businessPhone", 
+            "email", "fax", "prison", "url", "committeeCodes"]
+
+    def _map_result(self, result):
+        return RepresentativeInfo(
+            id=result[0],
+            dist=result[1], 
+            officeCode=result[2], 
+            districtNumber=result[3], 
+            designatorCode=result[4], 
+            firstName=result[5], 
+            middleInitial=result[6], 
+            lastName=result[7], 
+            suffix=result[8], 
+            commonlyUsedName=result[9], 
+            homeStreetAddress=result[10], 
+            homeCity=result[11], 
+            homeState=result[12], 
+            homeZipCode=result[13], 
+            homePhone=result[14], 
+            capitolStreetAddress=result[15], 
+            capitolCity=result[16], 
+            capitolPhone=result[17], 
+            room=result[18], 
+            roomNumber=result[19], 
+            committeesChaired=result[20], 
+            committeesViceChaired=result[21], 
+            rankingMember=result[22], 
+            committeeMember1=result[23], 
+            senatorRepresentative=result[24], 
+            party=result[25], 
+            title=result[26], 
+            gender=result[27], 
+            businessPhone=result[28], 
+            email=result[29], 
+            fax=result[30], 
+            prison=result[31], 
+            url=result[32], 
+            committeeCodes=result[33]
+        )
+
+
+    def get_id_from_name(self, name):
+        reps = self.getAll()
+        # I'm not sure if this is the cleanest and most robust way to do name lookups.
```

Comment:
You're right, it's definitely not, for a couple of reasons
1. You're paging every single record into memory when you're calling `self.getAll`, because that method returns a mapped row set, not an iterator. So not only is that potentially an expensive query, but you also need to touch every single row, even though you're really only after 1 record, at most
2. You're fetching waaay more data than you need from the database, because all this query does is return the `id` for a given name, so anything else that comes back in the result set (address, zip, committee data, etc) is all dropped on the floor 
3. This method returns on the first match, so if the table has the following reps `(Joe Bak, Jim Baker)`, and you called the method with the arg `Bak`, you might return `Jim Baker`, even though `Joe Bak` is probably a better response

This lookup is easy enough to just do with a sql query. The if statement you have below can be translated into a SQL query that uses the `LIKE` operator. LIKE is kinda like a fileglob, or a wildcard filter, so it's really useful for matching based on substrings

It'd be something like
```sql
SELECT rit.id FROM rep_info_table rit 
  WHERE rit.name LIKE '?%'
```

The `?` is going to be the parameter for the arg this method takes, and the `%` is the way that you tell the query to match anything that starts with `?`

So if name == `alex`, you'd get `LIKE alex%`, which would match `alex`, `alexbaluha`, `alex1180981247` etc, but would not match `foo alex`, `ale`, etc